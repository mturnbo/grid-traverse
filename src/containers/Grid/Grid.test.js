import React from 'react';
import { mount } from 'enzyme';
import GridComponent from './index';

const rows = 5;
const cols = 5;
const wrapper = mount(<GridComponent cols={cols} rows={rows} />);

test('Grid Component Renders', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Grid Component - rows prop', () => {
  expect(wrapper.prop('rows')).toEqual(rows);
});

test('Grid Component - cols prop', () => {
  expect(wrapper.prop('cols')).toEqual(cols);
});
