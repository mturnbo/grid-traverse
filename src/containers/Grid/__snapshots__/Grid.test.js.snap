// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Component Renders 1`] = `
<Grid
  cols={5}
  rows={5}
>
  <div
    className="grid"
  >
    <div
      className="row"
    >
      <Tile
        col={1}
        row={1}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          1
          :
          1
        </div>
      </Tile>
      <Tile
        col={2}
        row={1}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          1
          :
          2
        </div>
      </Tile>
      <Tile
        col={3}
        row={1}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          1
          :
          3
        </div>
      </Tile>
      <Tile
        col={4}
        row={1}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          1
          :
          4
        </div>
      </Tile>
      <Tile
        col={5}
        row={1}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          1
          :
          5
        </div>
      </Tile>
    </div>
    <div
      className="row"
    >
      <Tile
        col={1}
        row={2}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          2
          :
          1
        </div>
      </Tile>
      <Tile
        col={2}
        row={2}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          2
          :
          2
        </div>
      </Tile>
      <Tile
        col={3}
        row={2}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          2
          :
          3
        </div>
      </Tile>
      <Tile
        col={4}
        row={2}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          2
          :
          4
        </div>
      </Tile>
      <Tile
        col={5}
        row={2}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          2
          :
          5
        </div>
      </Tile>
    </div>
    <div
      className="row"
    >
      <Tile
        col={1}
        row={3}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          3
          :
          1
        </div>
      </Tile>
      <Tile
        col={2}
        row={3}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          3
          :
          2
        </div>
      </Tile>
      <Tile
        col={3}
        row={3}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          3
          :
          3
        </div>
      </Tile>
      <Tile
        col={4}
        row={3}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          3
          :
          4
        </div>
      </Tile>
      <Tile
        col={5}
        row={3}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          3
          :
          5
        </div>
      </Tile>
    </div>
    <div
      className="row"
    >
      <Tile
        col={1}
        row={4}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          4
          :
          1
        </div>
      </Tile>
      <Tile
        col={2}
        row={4}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          4
          :
          2
        </div>
      </Tile>
      <Tile
        col={3}
        row={4}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          4
          :
          3
        </div>
      </Tile>
      <Tile
        col={4}
        row={4}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          4
          :
          4
        </div>
      </Tile>
      <Tile
        col={5}
        row={4}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          4
          :
          5
        </div>
      </Tile>
    </div>
    <div
      className="row"
    >
      <Tile
        col={1}
        row={5}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          5
          :
          1
        </div>
      </Tile>
      <Tile
        col={2}
        row={5}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          5
          :
          2
        </div>
      </Tile>
      <Tile
        col={3}
        row={5}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          5
          :
          3
        </div>
      </Tile>
      <Tile
        col={4}
        row={5}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          5
          :
          4
        </div>
      </Tile>
      <Tile
        col={5}
        row={5}
      >
        <div
          className="tile"
          onClick={[Function]}
        >
          5
          :
          5
        </div>
      </Tile>
    </div>
  </div>
</Grid>
`;
